ipa-core/src/report.rs

```diff
@@ -587,9 +587,9 @@ mod test {
                 .collect(),
         };
 
-        let enc_key_registry = KeyRegistry::random(1, &mut rng);
+        let enc_key_registry = KeyRegistry::<KeyPair>::random(1, &mut rng);
         let enc_key_id = 0;
-        let dec_key_registry = KeyRegistry::random(1, &mut rng);
+        let dec_key_registry = KeyRegistry::<PrivateKeyOnly>::random(1, &mut rng);
```

Comment:
I think you should use `KeyPair` here too because `decrypt` takes `PrivateKeyRegistry` trait. That answers my question why you need `random` but also indicates that there is a change that needs to be made and `random` for `PrivateKeyOnly` should go away